File: JavaApplication1/src/commentaires/commentaire.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package commentaires;

import java.sql.Date;
import java.time.LocalDate;
import java.util.Objects;

/**
 *
 * @author ghmougui ons
 */
public class commentaire {
    private long id;
    private String text;
    private long iduser;
    private long idformation;
    private LocalDate date;
    private int evaluation;

   

   
    public int getEvaluation() {
        return evaluation;
    }

    public commentaire(long iduser,String text,  LocalDate date, int evaluation) {
        this.text = text;
        this.iduser = iduser;
        this.date = date;
        this.evaluation = evaluation;
    }

    public void setEvaluation(int evaluation) {
        this.evaluation = evaluation;
    }

    public commentaire(long id, String text, long iduser, long idformation, LocalDate date, int evaluation) {
        this.id = id;
        this.text = text;
        this.iduser = iduser;
        this.idformation = idformation;
        this.date = date;
        this.evaluation = evaluation;
    }
    public commentaire(){
        
    }
    

    public commentaire(long id, String text, long iduser, long idformation, LocalDate date) {
        this.id = id;
        this.text = text;
        this.iduser = iduser;
        this.idformation = idformation;
        this.date = date;
        
    }
    public commentaire(String text){
        this.text = text;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public long getIduser() {
        return iduser;
    }

    public void setIduser(long iduser) {
        this.iduser = iduser;
    }

    public long getIdformation() {
        return idformation;
    }

    public void setIdformation(long idformation) {
        this.idformation = idformation;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    @Override
    public String toString() {
        return "commentaire{" + "text=" + text + ", iduser=" + iduser + ", idformation=" + idformation + ", date=" + date + '}';
    }


    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final commentaire other = (commentaire) obj;
        if (this.id != other.id) {
            return false;
        }
        if (this.iduser != other.iduser) {
            return false;
        }
        if (this.idformation != other.idformation) {
            return false;
        }
        if (!Objects.equals(this.text, other.text)) {
            return false;
        }
        if (!Objects.equals(this.date, other.date)) {
            return false;
        }
        return true;
    }

    public commentaire(String text, int evaluation) {
        this.text = text;
        this.evaluation = evaluation;
    }
    

   
    

 

    
}

File: TestSymfony/src/Entity/Commentaire.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use DateTime;
use App\Repository\CommentaireRepository;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CommentaireRepository::class)]
class Commentaire
{
    
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    
    #[ORM\Column]
    private ?int $iduser = null;

    
    #[ORM\Column]
    private ? int $idformation = null;

    #[Assert\NotBlank(message:"Text can't be empty")]
    #[ORM\Column(length: 1000)]
    private ?string $text = null;

   
    #[ORM\Column]
    private ?DateTime $date = null;

    #[Assert\NotBlank(message:" Evaluation can't be null")]
    #[ORM\Column]
    private ?int $evaluation = null;

    #[ORM\ManyToOne( inversedBy: "commentaires")]
    #[ORM\JoinColumn(nullable: true)]
    private ?Formation $formation = null;


    public function getId(): ?string
    {
        return $this->id;
    }

    public function getIduser(): ?string
    {
        return $this->iduser;
    }

    public function setIduser(string $iduser): static
    {
        $this->iduser = $iduser;

        return $this;
    }

    public function getIdformation(): ?string
    {
        return $this->idformation;
    }

    public function setIdformation(string $idformation): static
    {
        $this->idformation = $idformation;

        return $this;
    }

    public function getText(): ?string
    {
        return $this->text;
    }

    public function setText(string $text): static
    {
        $this->text = $text;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getEvaluation(): ?int
    {
        return $this->evaluation;
    }

    public function setEvaluation(int $evaluation): static
    {
        $this->evaluation = $evaluation;

        return $this;
    }
    public function getFormation()
    {
        return $this->formation;
    }

    public function setFormation(Formation $formation = null)
    {
        $this->formation = $formation;

        return $this;
    }


}
